core/server/master/src/main/java/alluxio/master/metastore/rocks/RocksStore.java

```diff
@@ -142,7 +144,20 @@ private void createDb() throws RocksDBException {
     cfDescriptors.addAll(mColumnFamilyDescriptors);
     // a list which will hold the handles for the column families once the db is opened
     List<ColumnFamilyHandle> columns = new ArrayList<>();
-    mDb = RocksDB.open(mDbOpts, mDbPath, cfDescriptors, columns);
+    final TimeoutRetry retryPolicy = new TimeoutRetry(ROCKS_OPEN_RETRY_TIMEOUT, 100);
+    RocksDBException lastException = null;
+    while (retryPolicy.attempt()) {
+      try {
+        mDb = RocksDB.open(mDbOpts, mDbPath, cfDescriptors, columns);
+      } catch (RocksDBException e) {
+        // sometimes the previous terminated process's lock may not have been fully cleared yet
+        // retry until timeout to make sure that isn't the case
+        lastException = e;
+      }
+    }
+    if (mDb == null && lastException != null) {
```

Comment:
Is it possible for `mDb == null` and `lastException != null`?

Reply:
Yeah when RocksDB.open always throws a RocksDBException. lastException will not be null and mDB will never be populated.